sdk/go/pulumi-language-go/main.go

```diff
@@ -347,23 +384,34 @@ func (host *goLanguageHost) GetRequiredPlugins(ctx context.Context,
 		return nil, err
 	}
 
-	args := []string{"list", "-m", "-json", "-mod=mod", "all"}
+	gomod, err := host.loadGomod()
+	if err != nil {
+		// Don't fail if not using Go modules.
+		logging.V(5).Infof("GetRequiredPlugins: Error reading go.mod: %v", err)
+		return &pulumirpc.GetRequiredPluginsResponse{}, nil
+	}
+
+	args := make([]string, 0, len(gomod.Require)+3)
```

Comment:
I'm guessing it's extremely unlikely that anyone would run into [`ARG_MAX`](https://www.in-ulm.de/~mascheck/various/argmax/) limits.

Reply:
Yeah, it's unlikely but possible. I'll create an issue to follow up for that case in case it ever becomes a problem.

Reply:
FYI:

```
% docker runt ubuntu:latest xargs --show-limits
Your environment variables take up 110 bytes
POSIX upper limit on argument length (this system): 2094994
POSIX smallest allowable upper limit on argument length (all systems): 4096
Maximum length of command we could actually use: 2094884
Size of command buffer we are actually using: 131072
Maximum parallelism (--max-procs must be no greater): 2147483647
```

So the limit is about 2MB on my system.

Reply:
https://github.com/pulumi/pulumi/issues/12753